### PR TITLE
Further implementation of Server + some UI fixes

### PR DESCRIPTION
**Pull request by OrHaroni**

### CODE REVIEW

File: server/App.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ app.use(cors());
 // app.use(express.static('../public'));
 
 
-// app.use('/api/Users', userRoutes);
+app.use('/', userRoutes);
```
Comment (OrHaroni) [inline server/App.js]: add prefix of users

File: server/controllers/users.js [hunk 2]
```diff
@@ -1,9 +1,66 @@
...
 
-const getAllUsers = async () => {
-    return await userServices.getAllUsers();
+const getAllUsers = async (req, res) => {
+    const users = await userServices.getAllUsers();
+    if(users) {
+        res.status(200).json(users);
+    }
+    else{
+        res.status(500);
```
Comment (OrHaroni) [inline server/controllers/users.js]: add .json({})

File: server/services/users.js [hunk 3]
```diff
@@ -1,10 +1,59 @@
...
+    user.moneyAmount = 0;
+    userList.push(user);
+    return 0; //everything good
+}
+
+const addMoney = async (username, moneyAmount) => {
+    const user = userList.find(user => user.username === username);
+    if(user) {
+        console.log("Found user!");
+        console.log(user);
```
Comment (OrHaroni) [inline server/services/users.js]: Delete logs

File: src/Add_Money_Page/Add_Money_Page.js [hunk 4]
```diff
@@ -2,26 +2,44 @@ import React, { useRef } from 'react';
 import '../App.css';
 import Lobby, { sendSwal } from '../lobby/lobby.js';
 import { root } from '../index.js';
+import { addMoney } from '../serverCalls/Add_Money_page.js';
+
+function Add_Money_Page(props) {
+  console.log(props.user);
```
Comment (OrHaroni) [inline src/Add_Money_Page/Add_Money_Page.js]: delete log

File: src/login/login.js [hunk 5]
```diff
@@ -28,11 +28,14 @@ function Login() {
...
-      let isCorrectPassword = userExistsWithPassword(username.current.value, password.current.value);
-      if (!isCorrectPassword) {
+      let [user, status] = await userExistsWithPassword(username.current.value, password.current.value);
+      if (status !== 200) {
         sendSwal("Username or Password are incorrect", "warning");
       } else {
-        root.render(<Lobby />);
+        console.log("this is the connected user: ");
+        console.log(user);
+        sendSwal("Logged in Succesfully!", "success");
```
Comment (OrHaroni) [inline src/login/login.js]: delete this

Comment (OrHaroni) [reply src/login/login.js]: and logs

File: src/serverCalls/Add_Money_page.js [hunk 6]
```diff
@@ -1,3 +1,18 @@
...
+    const res = await fetch('http://localhost:8080/addMoney', {
+      'method' : 'post',
+      "headers" : {
+        'Content-Type': 'application/json',
+      },
+      'body': JSON.stringify(data)
+    });
+
+    let updatedUser = await res.json();
+    console.log(updatedUser);
```
Comment (OrHaroni) [inline src/serverCalls/Add_Money_page.js]: delete log

**Review by AdarKatz — APPROVED** [review]

**Pull request merged**